File: server/src/routes/groups.ts
import { Router } from 'express';
import { getCollection } from '../openproject/client.js';
import { mapGroup } from '../openproject/mappers.js';
import { asyncHandler } from '../middleware/error.js';
import type { Group } from '../openproject/mappers.js';

const router = Router();

router.get(
  '/',
  asyncHandler(async (_req, res) => {
    const raw = await getCollection<Record<string, any>>('/api/v3/groups');
    const groups: Group[] = raw.map(mapGroup);
    res.json(groups);
  }),
);

export default router;

File: server_py/app/routes/groups.py
from fastapi import APIRouter

from app.openproject.client import get_collection
from app.openproject.mappers import map_group
from app.openproject.schemas import Group


router = APIRouter(prefix="/api/groups")


@router.get("")
async def list_groups() -> list[Group]:
    raw = await get_collection("/api/v3/groups")
    return [map_group(item) for item in raw]
